### PR TITLE
551 update crate versions

### LINKED ISSUE #551: Update crate versions in dependencies
Same in Pearl: https://github.com/qoollo/pearl/issues/202

### PR DESCRIPTION
**Pull request by lostl1ght**
Closes #551
The following crates have severe breaking changes and require separate issues:

1. `axum` 0.5
2. `metrics` 0.20 (with this version the code compiles, however, I could not tell if [these](https://github.com/metrics-rs/metrics/blob/main/metrics/RELEASES.md#0180---2022-01-14) changes are too significant) 
2.1 `metrics-exporter-prometheus` 0.11
2.2 `metrics-util` 0.14
3. `tonic` 0.8
3.1 `tonic-build` 0.8
4. `prost` 0.11 in `bob-grpc`
5. `clap` 4.0 in `bob-apps`
6. ~~`crc` 3.0 may work or not but they do not provide changelog~~ it works
7. `jsonwebtoken` 8.1

### CODE REVIEW

**Comment by lostl1ght** [comment]
Updating `serde_yaml` to 0.9 breaks `bobd`. `bloom_filter_memory_limit` won't deserialize.
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: failed to parse
NodeConfig from yaml: error on yaml parsing: bloom_filter_memory_limit: invalid type:
string "8 GiB", expected a byte unit as an integer or string at line 59 column 28',
bob-apps/bin/bobd.rs:42:52
```
There is a not addressed related issue in `ubytes`: [link](https://github.com/SergioBenitez/ubyte/issues/7).
In their example they're also using 0.9.

File: bob-access/Cargo.toml [hunk 1]
```diff
@@ -9,12 +9,12 @@ axum = "0.4"
...
 http = "0.2"
-sha2 = "0.10.2"
+sha2 = "0.10.6"
 tower = "0.4"
 tonic = "0.6"
 log = "0.4"
 serde = {version = "1.0", features = ["derive"] }
 jsonwebtoken = "7.2"
-serde_yaml = "0.8"
+serde_yaml = "0.9"
```
Comment (ikopylov) [inline bob-access/Cargo.toml:18]: You wrote that using version `0.9` causes panic. Why did you update it?

File: bob-apps/Cargo.toml [hunk 2]
```diff
@@ -27,14 +27,14 @@ humantime = "2.1"
...
 metrics = { version = "0.17", features = ["std"] }
-mockall = "0.10"
-prost = "0.9"
+mockall = "0.11"
+prost = "0.11"
 ring = "0.16"
 serde = "1.0"
 serde_derive = "1.0"
-serde_yaml = "0.8"
+serde_yaml = "0.9"
```
Comment (ikopylov) [inline bob-apps/Cargo.toml:37]: Same here

**Comment by ikopylov** [comment]
> Updating `serde_yaml` to 0.9 in `bob-common` breaks `bobd`. `bloom_filter_memory_limit` won't deserialize.

Please, recheck this. Maybe it was fixed in the latest version

File: bob-backend/Cargo.toml [hunk 3]
```diff
@@ -15,21 +15,21 @@ bob-common = { path = "../bob-common", features = [] }
...
-mockall = "0.10"
+mockall = "0.11"
 parking_lot = "0.12.0"
-prost = "0.9"
+prost = "0.11"
 ring = "0.16"
 serde = "1.0"
 serde_derive = "1.0"
-serde_yaml = "0.8"
+serde_yaml = "0.9"
```
Comment (ikopylov) [inline bob-backend/Cargo.toml:32]: Same here

File: bob/Cargo.toml [hunk 4]
```diff
@@ -16,39 +16,39 @@ bob-access = { path = "../bob-access" }
...
-pipers = "1.0.0"
-prost = "0.9"
+mockall = "0.11.2"
+pipers = "1.0.1"
+prost = "0.11"
 ring = "0.16"
 serde = "1.0"
 serde_derive = "1.0"
-serde_yaml = "0.8"
+serde_yaml = "0.9"
```
Comment (ikopylov) [inline bob/Cargo.toml:37]: Same here

File: bob/Cargo.toml [hunk 5]
```diff
@@ -16,39 +16,39 @@ bob-access = { path = "../bob-access" }
...
+infer = "0.9"
 lazy_static = "1.4.0"
 libc = "0.2"
 log = "0.4"
 metrics = { version = "0.17", features = ["std"] }
-mockall = "0.10.2"
-pipers = "1.0.0"
-prost = "0.9"
+mockall = "0.11.2"
+pipers = "1.0.1"
```
Comment (ikopylov) [inline bob/Cargo.toml:32]: This crate was removed

**Comment by ikopylov** [comment]
@lostl1ght please, resolve merge conflicts

File: bob-access/Cargo.toml (hunk 1 above)
Comment (lostl1ght) [reply bob-access/Cargo.toml:18]: Only in bob-common.

Comment (lostl1ght) [reply bob-access/Cargo.toml]: I have unified `serde_yaml` version.

File: bob-apps/Cargo.toml (hunk 2 above)
Comment (lostl1ght) [reply bob-apps/Cargo.toml]: Same

File: bob-backend/Cargo.toml (hunk 3 above)
Comment (lostl1ght) [reply bob-backend/Cargo.toml]: Same

File: bob/Cargo.toml (hunk 4 above)
Comment (lostl1ght) [reply bob/Cargo.toml]: Same

File: bob/Cargo.toml (hunk 5 above)
Comment (lostl1ght) [reply bob/Cargo.toml]: Fixed

**Comment by lostl1ght** [comment]
@ikopylov I have resolved merge conflicts.

**Comment by lostl1ght** [comment]
Also, `crc` 3.0 works just fine.

**Review by ikopylov — APPROVED** [review]
LGTM
Pearl version should be updated before merging this

**Comment by ikopylov** [comment]
@lostl1ght please, resolve merge conflicts

**Comment by lostl1ght** [comment]
@ikopylov done

**Pull request merged**